doc/source/reference/functions/snippet.rst

```diff
@@ -0,0 +1,259 @@
+.. table_create Documents TABLE_NO_KEY
+.. column_create Documents content COLUMN_SCALAR Text
+.. table_create Terms TABLE_PAT_KEY ShortText --default_tokenizer TokenBigram  --normalizer NormalizerAuto
+.. column_create Terms documents_content_index COLUMN_INDEX|WITH_POSITION Documents content
+.. load --table Documents
+.. [
+.. ["content"],
+.. ["Groonga is a fast and accurate full text search engine based on inverted index. One of the characteristics of groonga is that a newly registered document instantly appears in search results. Also, groonga allows updates without read locks. These characteristics result in superior performance on real-time applications."],
+.. ["Groonga is also a column-oriented database management system (DBMS). Compared with well-known row-oriented systems, such as MySQL and PostgreSQL, column-oriented systems are more suited for aggregate queries. Due to this advantage, groonga can cover weakness of row-oriented systems."]
+.. ]
+
+``snippet`` extracts keywords from conditions specified in ``--query``
+and/or ``--filter`` automatically when you specify
+``default_open_tag`` option and ``default_close_tag`` and don't
+specify keywords. It's similar to :doc:`snippet_html`.
+
+The following example uses ``--query "fast performance"``. In this
+case, ``fast`` and ``performance`` are used as keywords.
+
+.. groonga-command
+.. include:: ../../example/reference/functions/snippet/usage_keywords_from_conditions.log
+.. select Documents \
+..   --output_columns 'snippet(content, \
+..                             { \
+..                                "default_open_tag": "[", \
+..                                "default_close_tag": "]" \
+..                             })' \
+..   --match_columns content \
+..   --query "fast performance"
+
+``--query "fast performance"`` matches to only the first record's
+content. This ``snippet`` extracts two text parts that include the
+keywords ``fast`` or ``performance`` and surrounds the keywords with
+``[`` and ``]``.
+
+The max number of text parts is 3 by default. You can change it by
+``max_n_results`` option:
+
+.. groonga-command
+.. include:: ../../example/reference/functions/snippet/usage_max_n_results.log
+.. select Documents \
+..   --output_columns 'snippet(content, \
+..                             { \
+..                                "default_open_tag": "[", \
+..                                "default_close_tag": "]", \
+..                                "max_n_results": 1 \
+..                             })' \
+..   --match_columns content \
+..   --query "fast performance"
+
+It returns only one snippet because ``"max_n_results": 1`` is specified.
+
+The max size of a text part is 200byte by default. The unit is bytes
+not characters. The size doesn't include inserted ``[`` and ``[``. You
+can change it by ``width`` option:
+
+.. groonga-command
+.. include:: ../../example/reference/functions/snippet/usage_max_n_results.log
```

Comment:
Is this example file correct?
Also, it seems that the `width` option isn't used at this usage in the built html.

Reply:
Good catch!